### ISSUE TITLE
Add config like requirePaddingNewlinesInBlocks but for functions

### ISSUE THREAD

**Issue by a-x-** [jscs-dev/node-jscs#1703]
`{ requirePaddingNewlinesInFunctionBodies: 3 }`

Invalid code:
```js
function(){
    foo();
    bar();
    baz();
    qux();
}
```
```
^--- Expected a padding newline after opening curly brace
```

**Upd**: Draft branch: https://github.com/jscs-dev/node-jscs/compare/master...bem-kit:require-padding-newlines-in-function-expressions

**Comment by markelog** [comment]
I assume you proposing the new rule?

**Comment by a-x-** [comment]
yeah, you're right

**Comment by markelog** [comment]
What valid/invalid code would be? Would you like to implement it?

**Comment by a-x-** [comment]
I updated description (see its for invalid case).

And see valid version below:

config: `{ requirePaddingNewlinesInFunctionBodies: 3 }`

valid code:
```js
function(){
    foo();
    bar();
    baz();
}
```

**Comment by markelog** [comment]
I'm not sure what you want here, limit amount of call expressions?

**Comment by a-x-** [comment]
Oh, no :)

Limit lines.

Now `requirePaddingNewlinesInBlocks` not works with functions' bodies. If it is expected,
i think would be nice to implement `requirePaddingNewlinesInFunctionBodies` rule for functions

**Comment by markelog** [comment]
Would you like to take a stab at it?

**Comment by a-x-** [comment]
valid code:
```js
function(){
    foo();
    var bar;
}
```

**Comment by a-x-** [comment]
Hmm, ok, i can try implement prototype.

**Comment by a-x-** [comment]
i found interesting place now: https://github.com/jscs-dev/node-jscs/blob/9b361a03047c9e9628f92f790d929671a57b53bc/lib/rules/require-padding-newlines-in-blocks.js#L157
`file.iterateNodesByType('BlockStatement', ...`
and i want find something `FunctionBody` NodeType

**Comment by a-x-** [comment]
there are at least following NodeTypes:
* FunctionDeclaration
* FunctionExpression

**Comment by a-x-** [comment]
as i understood, we should check `node.body.length` when NodeType is FunctionExpression

**Comment by a-x-** [comment]
Кажется всё просто:
Скопировал `./lib/rules/require-padding-newlines-in-blocks.js`
в `require-padding-newlines-in-function-expressions`

И поправил `BlockStatement` на `FunctionExpression` + jsdoc, ещё не проверял и тесты не правил.

Ветка: https://github.com/jscs-dev/node-jscs/compare/master...bem-kit:require-padding-newlines-in-function-expressions

**Comment by a-x-** [comment]
i added spec, but it failed

**Comment by markelog** [comment]
Please use English, could you try to sens us a PR?

**Comment by a-x-** [comment]
i'm not sure, that i can finish this work in short time.

What i did already:
> branch: https://github.com/jscs-dev/node-jscs/compare/master...bem-kit:require-padding-newlines-in-function-expressions

* cp  `./lib/rules/require-padding-newlines-in-blocks.js` into `./lib/rules/require-padding-newlines-in-function-expressions.js`
* replace `BlockStatement` to `FunctionExpression`
* update jsdoc
* update spec, but it, as i see, does not passing.

**Comment by markelog** [comment]
> i'm not sure, that i can finish this work in short time.

No hurry, just make a pr and we will figure out the rest

**Comment by a-x-** [comment]
Ok, i created now: https://github.com/jscs-dev/node-jscs/pull/1725

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
